src/mdp/sliding_puzzle.py

```diff
@@ -0,0 +1,101 @@
+import os
+import clingo
+import random
+from typing import Set, List
+
+from .markov_decision_procedure import MarkovDecisionProcedure
+
+class SlidingPuzzle(MarkovDecisionProcedure):
+
+    def __init__(self, state_initial: Set[str], state_static: Set[str]):
+
+        # TODO: check
+        # No discounting for Sliding
+        discount_rate = 1.0
+        file_name = 'sliding_puzzle.lp'
+
+        super().__init__(state_initial, state_static, discount_rate, file_name)
+
+class SlidingPuzzleBuilder:
+
+    def __init__(self, puzzle_size: int, missing_pieces: int, state_enumeration_limit: int = 9):
+
+        self.puzzle_size: int = puzzle_size
+        self.missing_pieces: int = missing_pieces
+        self.state_enumeration_limit: int = state_enumeration_limit
+
+        self.piece_terms: List[str] = [f'p{n}' for n in range(self.puzzle_size**2-self.missing_pieces)]
+
+        self.file_name: str  = 'sliding_puzzle_initial_states.lp'
+        self.file_path: str = os.path.join(os.path.dirname(os.path.abspath(__file__)), self.file_name)
+
+        if puzzle_size <= state_enumeration_limit:
+            self.all_states: List[Set[str]] = self._generate_all_states()
+
+        sample_mdp = self.build_mdp()
+        self.mdp_interface_file_path = sample_mdp.interface_file_path
+        self.mdp_problem_file_path = sample_mdp.problem_file_path
+        self.mdp_state_static = sample_mdp.state_static
+
+    def build_mdp(self):
+
+        # Use goal state in which all pieces are in numerical order
+        state_static = set()
+        positions = iter((i,j) for i in range(self.puzzle_size) for j in range(self.puzzle_size))
+        for piece in self.piece_terms:
```

Comment:
Just as a tip.. another way to do this would be:
```suggestion
        for piece, (x, y) in zip(self.piece_terms, positions):
            state_static.add(f'subgoal({piece},{x},{y})')
```
No need to change anything though. Your solution works fine ;-)